src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -1213,11 +1213,13 @@ private static bool HadLambdaConversionError(BindingDiagnosticBag diagnostics, A
                         new FormattedSymbol(UnwrapIfParamsArray(parameter, isLastParameter), SymbolDisplayFormat.CSharpErrorMessageNoParameterNamesFormat));
                 }
             }
-            else if (refArg != refParameter && !(refArg == RefKind.None && refParameter == RefKind.In))
+            else if (refArg != refParameter &&
+                !(refArg == RefKind.None && refParameter == RefKind.In) &&
+                !(refParameter == RefKind.RefReadOnlyParameter && refArg is RefKind.None or RefKind.Ref or RefKind.In))
```

Comment:
>(refParameter == RefKind.RefReadOnlyParameter && refArg is RefKind.None or RefKind.Ref or RefKind.In) [](http://example.com/codeflow?start=17&length=101)

Do we have tests demonstrating impact of this condition?

Reply:
Yes, `RefReadonlyParameter_WrongType` - wrong error would be reported without this condition ("wrong ref" instead of "wrong type").